### PR TITLE
Updated time/lat/lon validation

### PR DESCRIPTION
**Pull request by bhasin85**
#250

### CODE REVIEW

File: src/extension/wps/src/main/java/au/org/emii/wps/gogoduck/parameter/SubsetParameters.java [hunk 1]
```diff
@@ -71,18 +69,13 @@ public static SubsetParameters parse(String subset) {
...
+                calendarDateRange = CalendarDateRange.of(startTime, endTime);
+            } catch (IllegalArgumentException e) {
+                throw new GoGoDuckException(String.format("Invalid time format for subset: %s Valid time format example: TIME,2009-01-01T00:00:00.000Z,2009-12-25T23:04:00.000Z error: %s", subset, e.getMessage()));
             }
-            CalendarDate startTime = CalendarDate.parseISOformat("gregorian", timeRange.start);
-            CalendarDate endTime = CalendarDate.parseISOformat("gregorian", timeRange.end);
-            calendarDateRange = CalendarDateRange.of(startTime, endTime);
         }
 
         matcher = latLonPattern.matcher(subset);
```
Comment (jonescc) [inline src/extension/wps/src/main/java/au/org/emii/wps/gogoduck/parameter/SubsetParameters.java]: We've already proved that it matches this pattern by this point - why not remove this altogether like we have for time

File: src/extension/wps/src/main/java/au/org/emii/wps/gogoduck/parameter/SubsetParameters.java [hunk 2]
```diff
@@ -39,13 +39,11 @@ public Range getVerticalRange() {
...
-        Pattern timePattern = Pattern.compile("((1[7-9]|20)\\d\\d)-(0?[1-9]|1[012])-(0?[1-9]|[12][0-9]|3[01])T([0-9]+):([0-5]?[0-9]):([0-5]?[0-9]).([0-9]?[0-9]?[0-9]Z)");
+        int latLonCount = 0;
         Pattern latLonPattern = Pattern.compile("([+-]?\\d+\\.?\\d+)\\s*,\\s*([+-]?\\d+\\.?\\d+)");
         Map<String, ParameterRange> subsets = new HashMap<>();
         Matcher matcher;
 
-
         // Parse
         for (String part : subset.split(";")) {
             String[] subsetParts = part.split(",");
```
Comment (jonescc) [inline src/extension/wps/src/main/java/au/org/emii/wps/gogoduck/parameter/SubsetParameters.java]: Potential for subscript out of range error below.

File: src/extension/wps/src/test/java/au/org/emii/wps/gogoduck/parameter/SubsetParametersTest.java [hunk 3]
```diff
@@ -24,6 +24,14 @@ public void testParameterParsing() {
     }
 
     @Test
+    public void testValidTime() {
```
Comment (jonescc) [inline src/extension/wps/src/test/java/au/org/emii/wps/gogoduck/parameter/SubsetParametersTest.java]: Perhaps add a test for no decimal places as well which is what MARVL were providing.

**Comment by jonescc** [comment]
Tests not passing for some reason...

**Comment by bhasin85** [comment]
New [PR](https://github.com/aodn/geoserver-build/pull/252)

**Pull request closed without merging**